### PR TITLE
FST-10-36 - added oAuth 2.0 via Github and leaving comments only to users

### PR DESCRIPTION
**Pull request by iprisakaru**
Added oAuth via Github. For example when you want to insert a comment.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -13,6 +13,7 @@ flywayLocations += "db/migration"
 flywayBaselineOnMigrate := true
 
 libraryDependencies ++= Seq(
+  "org.apache.httpcomponents.client5" % "httpclient5" % "5.1",
```
Comment (iprisakaru) [inline build.sbt:16]: library for getting components of URI

File: src/main/scala/by/bsu/model/dao/UsersDAO.scala [hunk 2]
```diff
@@ -2,60 +2,70 @@ package by.bsu.model.dao
...
   import scala.concurrent.ExecutionContext.Implicits.global
 
   val LOGGER = Logger.getLogger(this.getClass.getName)
 
-  override type T = User
 
-  def insert(user: User): Future[Option[User]] = {
-    LOGGER.debug(s"Inserting user ${user.code}")
+  def insertGithub(user: User): Future[Option[User]] = {
+    LOGGER.debug(s"Inserting user ${user.ghLogin}")
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/model/dao/UsersDAO.scala:22]: Always give full names to variables. Your colleague could have troubles understanding what "gh" means
 githubLogin is much more clear

File: src/main/scala/by/bsu/utils/AuthService.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  protected implicit val executor: ExecutionContext = system.dispatcher
+
+
+  def getGHToken(code: String): Future[Option[String]] = {
+
+    val accessTokenFieldName = "access_token"
+
+    val request = HttpRequest(method = HttpMethods.POST,
+      uri = s"https://github.com/login/oauth/access_token?client_id=5c481d41c1df3a0cc298&client_secret=c674651120f1a7e4431ecca404f273552e3c18cd&code=$code")
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/AuthService.scala:39]: Do you think it makes sense to move github API urls to configuration?
I imagine a need to replace real URL with the mocked one in some circumstances

File: src/main/scala/by/bsu/utils/AuthService.scala [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    val responseFut = Http(system).singleRequest(request)
+    val tokenFut = responseFut.map(_.entity.toStrict(5 seconds)).flatMap(data => data.map(_.data.utf8String))
+    val uriFut = tokenFut.map(data => s"?$data")
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/AuthService.scala:42]: what is "fut"?

File: src/main/scala/by/bsu/utils/AuthService.scala [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    val params = uriFut.map(data => URLEncodedUtils.parse(new URI(data), Charset.forName("UTF-8")))
+
+    val accessToken = params.map(_.get(0)).map(data => {
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/AuthService.scala:45]: why 0?

File: src/main/scala/by/bsu/utils/AuthService.scala [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      if (data.getName == accessTokenFieldName) Option(data.getValue)
+      else None
+    })
+
+    accessToken
+  }
+
+  def getGitHubIdNameByToken(accessToken: String): Future[Option[String]] = {
+    val headerAuth = RawHeader("Authorization", s"Bearer $accessToken")
+    val request = HttpRequest(HttpMethods.GET, uri = "https://api.github.com/user").addHeader(headerAuth)
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/AuthService.scala:55]: Move URI to configuration

File: src/main/scala/by/bsu/utils/AuthService.scala [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      }
+      else None
+    }).map(HelpFunctions.fOption(_)).flatten.map(_.flatten)
+
+    githubUserByToken.map(_.flatMap(_.id)).map(_.map(_.toString))
+  }
+
+  def getGoogleIdEmailByToken(accessToken: String): Future[Option[String]] = {
+    val headerAuth = RawHeader("Authorization", s"Bearer $accessToken")
+    val request = HttpRequest(HttpMethods.GET, uri = "https://www.googleapis.com/oauth2/v2/userinfo").addHeader(headerAuth)
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/AuthService.scala:74]: Move to config

File: src/main/scala/by/bsu/utils/GithubUserService.scala [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package by.bsu.utils
+
+import by.bsu.model.dao.UsersDAO
+import by.bsu.model.repository.User
+
+class GithubUserService(githubUsersDAO: UsersDAO) {
+
+  def checkGithubUserByNode(githubUser: User) ={
```
Comment (alexander-belyaev) [inline src/main/scala/by/bsu/utils/GithubUserService.scala:8]: formatting

**Review by alexander-belyaev — CHANGES_REQUESTED** [review]

**Review by alexander-belyaev — APPROVED** [review]

**Pull request merged**